spec/models/user_spec.rb

```diff
@@ -0,0 +1,44 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  # let!(:user1) { User.create(name: 'John', posts_counter: 3) }
+  # let!(:user2) { User.create(name: 'Jane', posts_counter: 1) }
+  # let!(:user3) { User.create(name: 'Alice', posts_counter: 2) }
+
+  subject { User.new(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.', posts_counter: 0) }
+  before { subject.save }
+
+  it('name should be present') do
+    subject.name = nil
+    expect(subject).not_to be_valid
+  end
+
+  it('posts_counter should be a number') do
+    subject.posts_counter = nil
+    expect(subject).not_to be_valid
+  end
+
+  it('posts_counter should be greater than or equal to 0') do
+    subject.posts_counter = -1
+    expect(subject).not_to be_valid
+  end
+
+  it('posts_counter should be equal to 0') do
+    subject.posts_counter = 0
+    expect(subject).to be_valid
+  end
+
+  it('posts_counter should be greater than 0') do
+    subject.posts_counter = 3
+    expect(subject).to be_valid
+  end
+
+  it 'returns the three most recent users' do
+    user1 = User.create(name: 'John', photo: 'https://example.com/photo1.jpg', bio: 'Bio 1', posts_counter: 3)
+    user2 = User.create(name: 'Jane', photo: 'https://example.com/photo2.jpg', bio: 'Bio 2', posts_counter: 1)
+    user3 = User.create(name: 'Alice', photo: 'https://example.com/photo3.jpg', bio: 'Bio 3', posts_counter: 2)
+
+    recent_users = User.recent_three.to_a
+    expect(recent_users).to match_array([user1, user3, user2])
+  end
```

Comment:
- You have done great so far :+1: But there are a few requirements you need to meet before moving to the next milestone. :+1: 

- [ ] All your tests are passing correctly, and your tests seem to cover all the validations and methods, but this last test in the user model is not correct. 
The test is meant to be on a `three_most_recent_post` method in the user's model. Looking at the model, it seems the method you implemented there initially is a method that returns `3 recent users` but that's not the method that was in the requirement, the correct method is `A method that returns the 3 most recent posts for a given user.`. (this should have been spotted and mentioned by the previous reviewer of that milestone.).

I'll suggest you go to your user model and implement the correct method, then return to your test file and rewrite the test for this new correct method. :+1: :100: :+1: Here's a screenshot from the requirement of that milestone. :+1: 
- - Screenshot reference:
![Selection_024](https://github.com/wacoo/blog-front-ruby-rails/assets/107632780/3e47c6bb-0d33-4e06-8e18-7f75c8c634ec)